### PR TITLE
Adding some support for Arch Linux agent

### PR DESCRIPTION
**Pull request by avielw**
|Related issue|
#593

## Description
This commit should allow building Arch packages.

I've added a building script, Dockerfile and SPECS for the current version (4.2.0)

## Tests

- Build the package in any supported platform
  - [ ] Linux
  - [ ] Windows
  - [ ] macOS
  - [ ] Solaris
  - [ ] AIX
  - [ ] HP-UX
- [ ] Package installation
- [ ] Package upgrade
- [ ] Package downgrade
- [ ] Package remove
- [ ] Package install/remove/install
- [ ] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [ ] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [ ] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] Package install/remove/install
  - [ ] Package install/purge/install
  - [ ] Check file permissions after installing the package
- Tests for macOS
  - [ ] Test the package from macOS Sierra to Mojave
- Tests for Solaris
  - [ ] Test the package on Solaris 10
  - [ ] Test the package on Solaris 11
  - [ ] Check file permissions on Solaris 11 template
- Tests for IBM AIX
  - [ ] `%files` section is correctly updated if necessary
  - [ ] Check the changes from IBM AIX 5 to 7

### CODE REVIEW

File: arch/SPECS/4.2.0/wazuh-agent/arch/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	cp etc/templates/config/sles/sca.files ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/sca/sles
+	cp etc/templates/config/sles/11/sca.files ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/sca/sles/11
+
+	cp etc/templates/config/suse/sca.files ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/sca/suse
+	cp etc/templates/config/suse/11/sca.files ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/sca/suse/11
+
+	cp etc/templates/config/ubuntu/sca.files ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/sca/ubuntu
+	cp etc/templates/config/ubuntu/12/04/sca.files ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/sca/ubuntu/12/04
+	cp etc/templates/config/ubuntu/14/04/sca.files ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/sca/ubuntu/14/04
+	cp etc/templates/config/ubuntu/16/04/sca.files ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/sca/ubuntu/16/04
```
Comment (okynos) [inline arch/SPECS/4.2.0/wazuh-agent/arch/PKGBUILD:150]: Remove unnecessary SCA files, let only generic ones.

File: arch/SPECS/4.2.0/wazuh-agent/arch/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    chmod +x ${srcdir}/wazuh/fix_owner.sh
+}
+
+package() {
+    install -Dm0644 ${srcdir}/wazuh.sysusers ${pkgdir}/usr/lib/sysusers.d/wazuh.conf
+    cd ${srcdir}/wazuh
+
+	# Copying systemd file
+	mkdir -p ${pkgdir}/usr/lib/systemd/system/
+	install -m 0644 src/systemd/wazuh-agent.service ${pkgdir}/usr/lib/systemd/system/
```
Comment (okynos) [inline arch/SPECS/4.2.0/wazuh-agent/arch/PKGBUILD:55]: The path in 4.2.0 of this file is `src/init/templates/wazuh-agent.template`

File: arch/SPECS/4.2.0/wazuh-agent/arch/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	# Copying install scripts to /usr/share
+	mkdir -p ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/
+	cp -p gen_ossec.sh ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/
+	cp -p add_localfiles.sh ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/
+    cp -p fix_owner.sh ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"
+
+	mkdir -p ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/src
+	cp -p src/VERSION ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/src/
+	cp -p src/REVISION ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/src/
+	cp -p src/LOCATION ${pkgdir}/"PARAM_INSTALLATION_SCRIPTS_DIR"/src/
```
Comment (okynos) [inline arch/SPECS/4.2.0/wazuh-agent/arch/PKGBUILD:81]: Removed on 4.2.0

**Review by okynos — CHANGES_REQUESTED** [review]
Hello and thanks for your contribution, to merge this PR we need some points to be checked:
- Update last 4.2.0 and 4.2.1 changes.
- Test package building process.

I got an error after the PKGBUILD fixes
```
==> Entering fakeroot environment...
==> Starting package()...
==> ERROR: A failure occurred in package().
    Aborting...
```

We will be glad to help you with the process.

**Comment by avielw** [comment]
@okynos I think I did all the changes you asked for and fixed the issue with the package building. Now it looks like everything works for version 4.2.0.

**Review by okynos — APPROVED** [review]
LGTM.

**Comment by okynos** [comment]
Thanks a lot for your contribution, we are always working to improve our product and our procedures.

Regards!

**Comment by avielw** [comment]
Hey @okynos,
I saw that in branch **4.2** all the specs were updated to version 4.2.2, so I updated it for arch too. Do you want to pull this new commit too?

**Comment by okynos** [comment]
Hello @avielw,

We will be glad if you open a new PR including such changes, thanks!

**Comment by avielw** [comment]
@okynos In case you've misted it:
https://github.com/wazuh/wazuh-packages/pull/873

**Pull request merged**